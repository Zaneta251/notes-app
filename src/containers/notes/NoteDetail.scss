@import "../../css/variables";

.note-detail-layout {
  background: #D7CCC8;
  width: 100%;
}

.note-detail {
  height: 100%;
  max-width: 100%;
  display: block;
  margin: 20px;
  background: black;
  background: ghostwhite;
  color: black;
  padding: 10px;
}

.action-buttons {
  text-align: right;
  position: absolute;
  right: 5px;
  margin: 5px;
}

.text-area {
  height: 90vh;
}

.itemButtons button {
  display: inline-block;
  margin-left: 4px;
}

.bsc-text {
  font-size: 200px;
  color: white;
  text-shadow: 3px 2px black;
  text-align: center;
  position: relative;
  top: 35%;
}