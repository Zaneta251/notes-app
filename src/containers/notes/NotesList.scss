@import "../../css/variables";

.note-list-layout {
  background: #001627;
  width: 400px;
  display: flex;
  flex-direction: column;
  cursor: pointer;
}

.note-list {
  height: 100%;
  max-width: 100%;
  margin: auto;
  display: block;
}

.note {
  height: 50px;
}

.scrollable-container {
  height: calc(100vh - 250px);
  display: flex;
  flex-direction: column;
  cursor: pointer;
  width: 100%;
  border-bottom: none;
}

.hoverable-list-item {
  -moz-transition: all 0.2s ease-in;
  -o-transition: all 0.2s ease-in;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  cursor: pointer;
  color: #EEEEEE;
}
.hoverable-list-item:hover {
  color: black;
  -moz-transition: all 0.2s ease-in;
  -o-transition: all 0.2s ease-in;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  background-color: #00ACFA;
}

.hoverable-shadow {
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
}

.hoverable-shadow:hover {
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.75);
}
