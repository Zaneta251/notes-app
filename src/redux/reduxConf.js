import {applyMiddleware, compose} from "redux";
import thunk from "redux-thunk";
import freeze from "redux-freeze";

const crashReporter = () => next => async action => {
  try {
    return await Promise.resolve(next(action));
  } catch (err) {
    // here can go any other automated error handling
    // - such as displaying error messages to user, if they come from BE
    // - or sending to some external logging tool
    console.error(err);
    next({type: "ERROR"});
  }
};

const middlewares = [];
middlewares.push(crashReporter);
middlewares.push(thunk);

// freeze is only for development env
if (process.env.NODE_ENV !== "production") {
  middlewares.push(freeze);
}

let middleware = applyMiddleware(...middlewares);
// add the redux dev tools
if (process.env.NODE_ENV !== "production" && window.devToolsExtension) {
  middleware = compose(middleware, window.devToolsExtension());
}

export default middleware;
