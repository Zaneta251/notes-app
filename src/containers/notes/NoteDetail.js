import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Button, Input } from "antd";
import "./NoteDetail.scss";
import { addNewNote, setSelectedNote, setCreateFlag, deleteNote, editNote } from '../main/actions';
import { strings } from "../translate/strings";

const { TextArea } = Input;

class NoteDetail extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      newNote: "",
      editedNote: ""
    };
  }

  editNote = note => {
    this.setState({
      editedNote: note.title
    });
    this.props.setCreateFlag(true);
  };

  addNewNote = () => {
    if (this.state.newNote.length !== 0) {
      this.props.addNewNote(this.state.newNote);
    }
  }

  onCancel = () => {
    this.props.setCreateFlag(false);
    this.setState({ newNote: "", editedNote: "" })
  }

  deleteNote = () => {
    this.props.deleteNote(this.props.selectedNote.id);
    this.props.setSelectedNote(null);
  }

  saveEditedNote = () => {
    if (this.state.editedNote !== this.props.selectedNote.title) {
      this.props.editNote(this.props.selectedNote.id);
    }
  }

  render() {
    const { selectedNote, createFlag } = this.props;

    return (
      <div className="note-detail-layout">
        {selectedNote ?
          createFlag ?
            <div className="text-area">
              <TextArea
                style={{ height: "90%", width: "100%" }}
                placeholder="Add new note ... "
                value={this.state.editedNote}
                onChange={e => this.setState({ editedNote: e.target.value })}
              />
              <div className="action-buttons">
                <Button onClick={this.onCancel}>{strings.cancel}</Button>
                <Button onClick={this.deleteNote}>{strings.delete}</Button>
                <Button onClick={this.saveEditedNote}>{strings.save}</Button>
              </div>
            </div>
            :
            <div>
              <div className="note-detail">{selectedNote.title}</div>
              <div className="action-buttons">
                <Button onClick={() => this.props.setSelectedNote(null)}>{strings.cancel}</Button>
                <Button onClick={this.deleteNote}>{strings.delete}</Button>
                <Button onClick={() => this.editNote(selectedNote)}>{strings.edit}</Button>
              </div>
            </div>
          :
          createFlag ?
            <div className="text-area">
              <TextArea
                style={{ height: "95%", width: "100%" }}
                placeholder="Add a new note ... "
                value={this.state.newNote}
                onChange={e => this.setState({ newNote: e.target.value })}
              />
              <div className="action-buttons">
                <Button onClick={this.onCancel}>{strings.cancel}</Button>
                <Button onClick={this.addNewNote}>{strings.add}</Button>
              </div>
            </div>
            :
            <div className="bsc-text">{strings.BSC}</div>
        }
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    selectedNote: state.main.selectedNote,
    notes: state.main.notesList,
    createFlag: state.main.newNote
  };
};

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setCreateFlag,
      setSelectedNote,
      addNewNote,
      deleteNote,
      editNote
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NoteDetail);
