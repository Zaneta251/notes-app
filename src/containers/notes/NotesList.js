import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Col, Row, Button } from "antd";
import { strings } from "../translate/strings";
import "./NotesList.scss";
import { loadNotesList, setSelectedNote, setCreateFlag } from '../main/actions';

class NotesList extends React.Component {
  openDetail = note => {
    this.props.setSelectedNote(note);
    this.props.setCreateFlag(false);
  }

  render() {
    return (
      <div className="note-list-layout">
        <div style={{ overflowY: "auto", maxHeight: "100vh" }} className="scrollable-container">
          {this.props.notes.map(note => (
            <div key={`area-${note.id}`}>
              <div className="hoverable-list-item note"
                onClick={() => this.openDetail(note)}>
                <Row>
                  <Col span={15}>
                    <div style={{ padding: "2px" }}>
                      {strings.note}: {note.title}
                    </div>
                  </Col>
                </Row>
              </div>
            </div>
          ))}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    notes: state.main.notesList
  };
};

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      loadNotesList,
      setSelectedNote,
      setCreateFlag
    },
    dispatch
  );

export default connect(
  mapStateToProps, mapDispatchToProps
)(NotesList);

