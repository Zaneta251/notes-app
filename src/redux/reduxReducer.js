import { combineReducers } from "redux";
import {
  main
} from "../containers/main/reducers";

const rootReducer = combineReducers({
  main
});

export default rootReducer;
