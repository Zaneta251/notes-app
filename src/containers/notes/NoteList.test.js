import NotesList from "./NotesList";
import NoteDetail from "./NoteDetail";
import { Provider } from "react-redux";
import { combineReducers } from "redux";
import {
  main
} from "../main/reducers";
import { createStore } from 'redux';

const myReducers = combineReducers({
  main: main
});

const store = createStore(
  myReducers
);

it("renders correctly", () => {
  const wrapper = shallow(
    <NotesList />
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly", () => {
  const wrapper = shallow(
    <NoteDetail />
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly again", () => {
  const wrapper = render(
    <Provider store={store}>
      <NotesList />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly again", () => {
  const wrapper = render(
    <Provider store={store}>
      <NoteDetail />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
