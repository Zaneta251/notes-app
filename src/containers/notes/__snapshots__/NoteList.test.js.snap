// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`renders correctly 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`renders correctly again 1`] = `
<div
  class="note-list-layout"
>
  <div
    class="scrollable-container"
    style="overflow-y:auto;max-height:100vh"
  />
</div>
`;

exports[`renders correctly again 2`] = `
<div
  class="note-detail-layout"
>
  <div
    class="bsc-text"
  >
    BSC
  </div>
</div>
`;

exports[`renders correctly again 3`] = `
<div
  className="note-detail-layout"
>
  <div
    className="bsc-text"
  >
    BSC
  </div>
</div>
`;
